### ISSUE TITLE
filters.laplace result has the wrong sign

### ISSUE THREAD

**Issue by Matthias-Schroeter** [scikit-image/scikit-image#7357]
### Description:

The purpose of a Laplace filter is to compute an approximation for the second derivative. Which means that at the position of a maximum, it should give us back a negative number. That is e.g. the behavior of the scipy implementation of a Laplace filter. The filter matrix in skimage is inverted.
Fixing this bug will break the workarounds some people might have used in the past (i.e. multiplying the skimage result with minus 1). 

### Way to reproduce:

import numpy as np
import skimage as ski

from scipy import signal
maximum = np.outer(signal.windows.gaussian(5, 1)*10, signal.windows.gaussian(5,1))*10
np.set_printoptions(precision = 1)
print ('a 2D Gaussian maximum:\n',maximum)

laplace_result= ski.filters.laplace(maximum)
print('\nLaplace filter result:\n',laplace_result)

### as can be seen by a comparison with the scipy implementation
from scipy import ndimage

A = np.zeros((3,3))
A[1,1] = 1

laplace_filter_scipy = ndimage.laplace(A)
laplace_filter_skimage= ski.filters.laplace(A)

print('scipy:\n',laplace_filter_scipy)
print('\nskimage:\n',laplace_filter_skimage)

### Version information:

```Shell
3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0]
Linux-6.7.9-1-default-x86_64-with-glibc2.39
scikit-image version: 0.22.0
numpy version: 1.26.0
```

**Comment by stefanv** [comment]
I don't think the Laplacian kernel is wrong, per se; there are two conventions (positive vs negative peak), and skimage uses the slightly less common one. Maybe @glemaitre remembers why he chose that variant :)

**Comment by Matthias-Schroeter** [comment]
I believe most mathematicians would agree with the second derivative argument. Wikipedia only knows the center negative version: https://en.wikipedia.org/wiki/Discrete_Laplace_operator#Image_processing

As suggested by mcor under in the commit of pitkajuh a change in naming might be needed to alert users.

**Comment by stefanv** [comment]
I'm not arguing that there is no difference, of course :) Just saying you find both in the wild. Not sure agter so many years it's worth changing the sign, but it is something we could put on the skimage2 list, where we will be breaking a bunch of backward compatibility anyway. /cc @lagru

**Comment by glemaitre** [comment]
As far I recall when looking at the original PR, I reuse the `skimage.restoration.uft.laplacian` to define it.

https://github.com/scikit-image/scikit-image/blob/a160f384523ac70173d069954b7084bee79fb68e/skimage/restoration/uft.py#L411-L451

When it comes if this is right or wrong, I would check the "Digital Image Processing" reference and I get both formulation:

![IMG_1630](https://github.com/user-attachments/assets/89577b58-ad0f-44d8-9d27-082bd5069149)

**Comment by lagru** [comment]
Hmm, SciPy's definition seems sensible without concern for backwards compatibility. However, with the default now, my gut feeling right now is to leave it as is. The definition we are using seems not entirely exotic and confusing. However, we should definitely document this definition in the docstrings of both affected functions.

**Comment by lagru** [comment]
That said, I could live with us switching the definition for skimage2.

**Comment by Matthias-Schroeter** [comment]
I would like to strongly  encourage switching to the mathematically correct version for skimage2.

I am teaching image processing and computer vision at 2 different universities. The reason I found this issue is that the edge sharpening with a Laplace filter technique (described in the screenshots from Burger and Burge below) did not work with skimage.  It took me a moment to figure out why. Now I can of course fix this by switching the sign. But the math behind a second derivative filter is rather undisputable.

![sharpening1](https://github.com/user-attachments/assets/497dd4ba-7de3-4db5-9985-615a8544e93e)
![sharpening2](https://github.com/user-attachments/assets/3204f1d9-b92f-4b3e-8b3f-435ccee78060)
![sharpening3](https://github.com/user-attachments/assets/9950b941-e984-4254-8e94-81fd330c65d5)

**Comment by stefanv** [comment]
We may also want to provide larger kernels, while we're at it, if that'd be useful; could be a later addition.

**Comment by glemaitre** [comment]
I was just reading a bit around and the filters are named positive Laplacian and negative Laplacian. In the case of @Matthias-Schroeter and the image sharpening case, the formulation only works with the positive Laplacian, obviously.

Thinking a bit more, would it be against scikit-image API principles to introduce a parameter allowing to switch representations, e.g., `sign={"negative", "positive"}` / `sign={-1, 1}`. This would acknowledge that both representations exist and correct and allow for the case of image sharpening. Through deprecation, you could even change the default if desired.

**Comment by stefanv** [comment]
That may be helpful to make what is detected more understandable, even if it is rather annoying to have a flag that, basically, just indicates whether or not to multiply by -1. `negative`/`positive`, also sometimes called `inner`/`outer` etc. And then we just change the default in skimage2, which we do for a lot of other functions also.

**Comment by stefanv** [comment]
> So at the end, it only looks like they are just convention and both positive and negative Laplacian will have their use cases and it is just a matter of taste (which reinforce my thought about providing the option to the user :)).

Agreed that they both have use-cases, but it's a bit annoying on our part to implement something named after a very specific mathematical operator with negative-of-that-operator.

**Comment by glemaitre** [comment]
> but it's a bit annoying on our part to implement something named after a very specific mathematical operator with negative-of-that-operator.

Yep agreed.

**Comment by lagru** [comment]
Okay, I added it as a TODO in our [skimage2 list](https://github.com/scikit-image/scikit-image/wiki/API-changes-for-skimage2#skimagefilters).

Not too fond of having a switch parameter here. I'd rather change the behavior in skimage2 and document this issue in the docstring.

**Comment by stefanv** [comment]
That works, thanks.
